File: app/src/main/java/com/example/learnwordstrainer/ui/activities/BubbleSettingsActivity.java
package com.example.learnwordstrainer.ui.activities;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import com.example.learnwordstrainer.databinding.ActivityBubbleSettingsBinding;
import com.example.learnwordstrainer.viewmodels.BubbleSettingsViewModel;
import com.example.learnwordstrainer.viewmodels.BubbleViewModel;

public class BubbleSettingsActivity extends AppCompatActivity {
    private ActivityBubbleSettingsBinding binding;
    private BubbleSettingsViewModel viewModel;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityBubbleSettingsBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        viewModel = new ViewModelProvider(this).get(BubbleSettingsViewModel.class);

        binding.fabBack.setOnClickListener(v -> finish());


    }
}

File: app/src/main/java/com/example/learnwordstrainer/ui/activities/BubbleSettingsActivity.kt
package com.example.learnwordstrainer.ui.activities

import BubbleSettingsScreen
import BubbleSettingsViewModel
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.viewModels
import androidx.compose.runtime.getValue
import androidx.lifecycle.compose.collectAsStateWithLifecycle
import com.example.learnwordstrainer.ui.theme.LearnWordsTrainerTheme

class BubbleSettingsActivity : ComponentActivity() {

    private val viewModel: BubbleSettingsViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContent {
            val uiState by viewModel.uiState.collectAsStateWithLifecycle()

            LearnWordsTrainerTheme {
                BubbleSettingsScreen(
                    uiState = uiState,
                    onBackPressed = { finish() },
                    onBubbleEnabledChange = viewModel::setBubbleEnabled,
                    onBubbleSizeChange = viewModel::setBubbleSize,
                    onTransparencyChange = viewModel::setBubbleTransparency,
                    onVibrationEnabledChange = viewModel::setVibrationEnabled,
                    onAboutBubbleClick = {},
                    onAutoHideClick = {})
            }
        }
    }
}